include/extractor/extraction_way.hpp

```diff
@@ -50,6 +65,20 @@ struct ExtractionWay
     void set_backward_mode(const TravelMode m) { backward_travel_mode = m; }
     TravelMode get_backward_mode() const { return backward_travel_mode; }
 
+    // wrappers to allow assigning nil (nullptr) to string values
+    void SetName(const char* value) { detail::maybeSetString(name, value); }
```

Comment:
On second thought: since we want to have it less error prone in the profiles, having these setters inside of OSRM is the more secure version to do. Otherwise we could still run into problems when someone forgets to use the function I proposed.